components/wrap/WrapModal.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import Modal from "react-bootstrap/Modal";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import { ethers } from "ethers";
+import { ethxABI, NETWORK_ID, PAYMENT_TOKEN_ADDRESS } from "../../lib/constants";
+import { getETHBalance, getETHxBalance } from "../../lib/getBalance";
+
+function WrapMpdal({ account, signer, show, handleClose }: any) {
+  const [ETHBalance, setETHBalance] = React.useState<string | undefined>();
+  const [ETHxBalance, setETHxBalance] = React.useState<string | undefined>();
+  const [isWrapping, setIsWrapping] = React.useState<boolean>(false);
+  const [isBalanceInsufficient, setIsBalanceInsufficient] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    (async () => {
+      const provider = new ethers.providers.InfuraProvider(
+        NETWORK_ID,
+        process.env.NEXT_PUBLIC_INFURA_ID
+      );
+
+      const ethBalance = await getETHBalance(provider, account);
+
+      const ethxBalance = await getETHxBalance(provider, account);
+
+      if (isMounted) {
+        setETHBalance(ethBalance);
+        setETHxBalance(ethxBalance);
+      }
+    })();
+
+    return () => {
+      isMounted = false;
+    };
+  }, []);
+
+  const wrapETH = async (amount: string) => {
+    const provider = new ethers.providers.InfuraProvider(
```

Comment:
Both the provider and signer can be reused from the `library` here https://github.com/Geo-Web-Project/cadastre/blob/develop/pages/index.tsx#L57

We should probably include this as a prop. This should be easier after resolving the merge conflicts from `develop`, since I reorganized how props are passed.